pyannote/audio/applications/change_detection.py

```diff
@@ -0,0 +1,411 @@
+                # apply peak detection
+                hypothesis = peak.apply(predictions[uri])
+                # compute purity and coverage
+                purity[i](reference, hypothesis)
+                coverage[i](reference, hypothesis)
+
+        TEMPLATE = '{alpha:.2f} {purity:.1f}% {coverage:.1f}%'
+        res = []
+        for i, a in enumerate(alphas):
+            p = 100 * abs(purity[i])
+            c = 100 * abs(coverage[i])
+            print(TEMPLATE.format(alpha=a, purity=p, coverage=c))
+            res.append((a,p,c))
+        return res
+
+
+    res = objective_function(nb_epoch-1)
+
+    with open(store_dir + '/res.yml', 'w') as fp:
+        yaml.dump(res, fp, default_flow_style=False)
+
+
+def apply(protocol, train_dir, store_dir, threshold, subset='development'):
+    #os.makedirs(store_dir)
+
+    # -- LOAD MODEL --
+    architecture_yml = train_dir + '/architecture.yml'
+    WEIGHTS_H5 = train_dir + '/weights/{epoch:04d}.h5'
+    nb_epoch = 0
+    while True:
+        weights_h5 = WEIGHTS_H5.format(epoch=nb_epoch)
+        if not os.path.isfile(weights_h5):
+                break
+        nb_epoch += 1
+    config_dir = os.path.dirname(os.path.dirname(train_dir))
+    config_yml = config_dir + '/config.yml'
+    with open(config_yml, 'r') as fp:
+            config = yaml.load(fp)
+
+    # -- FEATURE EXTRACTION --
+    feature_extraction_name = config['feature_extraction']['name']
+    features = __import__('pyannote.audio.features',
+                                                fromlist=[feature_extraction_name])
+    FeatureExtraction = getattr(features, feature_extraction_name)
+    feature_extraction = FeatureExtraction(
+            **config['feature_extraction'].get('params', {}))
+
+    # -- SEQUENCE GENERATOR --
+    duration = config['sequences']['duration']
+    step = config['sequences']['step']
+
+    def saveSeg(filepath,filename,chn,segmentation):
+        f = open(filepath,'w')
+        for idx, val in enumerate(segmentation):
+                line = filename+' '+str(idx)+' '+str(chn)+' '+str(int(val[0]*100))+' '+str(int(val[1]*100-val[0]*100))+'\n'
+                f.write(line)
+        f.close()
+
+    def get_aggregation(epoch):
+        weights_h5 = WEIGHTS_H5.format(epoch=epoch)
+        sequence_labeling = SequenceLabeling.from_disk(
+                architecture_yml, weights_h5)
+
+
+        aggregation = SequenceLabelingAggregation(
+                sequence_labeling, feature_extraction,
+                duration=duration, step=step)
+
+        return aggregation
+
+
+    filepath = store_dir+'/'+str(threshold) +'/'
+    os.makedirs(filepath)
+
+    aggregation = get_aggregation(nb_epoch-1)
+
+    predictions = {}
+    for dev_file in getattr(protocol, subset)():
+        uri = dev_file['uri']
+        predictions[uri] = aggregation.apply(dev_file)  
+
+    # initialize peak detection algorithm
+    peak = Peak(alpha=threshold, min_duration=2.5)
```

Comment:
Maybe we can also add a `--min-duration` option to set this value.
The default can still be `2.5` if this is the one used in the paper...